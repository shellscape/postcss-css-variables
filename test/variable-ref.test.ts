import { fixture, process } from './helpers';

const tests = {
  'should work with local variables that reference other variables with at-rule changing the value':
    'variable-reference-other-variable-media-query2',
  'should work with variable with calc-expression that reference other variables':
    'variable-with-calc-expression-reference-other-variable',
  'should work with variables that reference other variables': 'variable-reference-other-variable',
  'should work with variables that reference other variables with at-rule changing the value':
    'variable-reference-other-variable-media-query1'
};

describe('variable-ref', async () => {
  test.each(Object.entries(tests))('%s', async (_desc, file) => {
    const css = await fixture(`variable-ref/${file}`);
    const result = await process(css);
    expect(result).toMatchSnapshot();
  });
});
