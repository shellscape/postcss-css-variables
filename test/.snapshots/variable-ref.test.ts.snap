// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`variable-ref > should work with local variables that reference other variables with at-rule changing the value 1`] = `".foo{width:150px}@media (max-width: 960px){.foo{width:300px}}"`;

exports[`variable-ref > should work with variable with calc-expression that reference other variables 1`] = `".foo{width:calc(50px + 150px)}@media (max-width: 960px){.foo{width:calc(50px + 300px)}}"`;

exports[`variable-ref > should work with variables that reference other variables 1`] = `".box-foo{width:calc(150px + 15vw)}"`;

exports[`variable-ref > should work with variables that reference other variables with at-rule changing the value 1`] = `".foo{width:150px}@media (max-width: 960px){.foo{width:300px}}"`;
